src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/items/SoulboundRune.java

```diff
@@ -0,0 +1,117 @@
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.Research;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Objects.handlers.ItemDropHandler;
+import me.mrCookieSlime.Slimefun.Setup.Messages;
+import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.PlayerProfile;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class SoulboundRune extends SimpleSlimefunItem<ItemDropHandler> {
+    private Research research = null;
+
+    public SoulboundRune(Category category, ItemStack item, String id, RecipeType type, ItemStack[] recipe) {
+        super(category, item, id, type, recipe);
+    }
+
+    @Override
+    public ItemDropHandler getItemHandler() {
+        return (e, p, i) -> {
+            ItemStack item = i.getItemStack();
+            if (SlimefunManager.isItemSimiliar(item, SlimefunItems.RUNE_SOULBOUND, true)) {
+                if (!PlayerProfile.fromUUID(p.getUniqueId()).hasUnlocked(research)) {
+                    Messages.local.sendTranslation(p, "messages.not-researched", true);
+                    return true;
+                }
+
+                Bukkit.getScheduler().scheduleSyncDelayedTask(SlimefunPlugin.instance, () -> {
+                    // Being sure the entity is still valid and not picked up or whatsoever.
+                    if (!i.isValid()) return;
+
+                    Location l = i.getLocation();
+                    Collection<Entity> entites = l.getWorld().getNearbyEntities(l, 1.5, 1.5, 1.5,
+                            entity -> entity.getType() == EntityType.DROPPED_ITEM &&
+                                    !SlimefunManager.isItemSimiliar(((Item) entity).getItemStack(), SlimefunItems.RUNE_SOULBOUND, true)
+                    );
+
+                    ItemStack ench = null;
+                    Item ent = null;
+                    // Collections do not have a #get method so we need to use a for loop.
+                    // We do not use streams for foreach loops as they are more resource consuming.
+                    for (Entity entity: entites) {
+                        ItemStack dropped = ((Item) entity).getItemStack();
+                        if (SlimefunManager.isItemSoulbound(dropped)) return;
+                        ench = ((Item) entity).getItemStack();
+                        ent = (Item) entity;
+                        break;
+                    }
+
+                    if (ench == null || ench.getAmount() == 1) {
+                        e.setCancelled(true);
+
+                        Item finalEnt = ent;
+                        ItemStack finalEnch = ench;
+
+                        ItemMeta enchMeta = finalEnch.getItemMeta();
+                        if (enchMeta == null) enchMeta = Bukkit.getItemFactory().getItemMeta(finalEnch.getType());
```

Comment:
This is redundant, geItemMeta() already performs this call if no ItemMeta has been found.

Reply:
Hmm didn't know that